### PR TITLE
Add some metadata

### PR DESCRIPTION
**Pull request by Bigaston**
I've added metadata for :
- platform
- social
- donate
- shop
- link
- person
- team
- guest
- book

And I've define some simple and complex type

### CODE REVIEW

**Comment by PofMagicfingers** [comment]
Wow thanks for that, I'll look into it 👍

**Comment by Bigaston** [comment]
Your welcome! I've edit some of your specification to use some global type I define ^^

File: podcast.xsd [hunk 1]
```diff
@@ -1,11 +1,9 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <xsd:schema
     xmlns:xsd="http://www.w3.org/2001/XMLSchema"
-    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
```
Comment (PofMagicfingers) [inline podcast.xsd:4]: XSI allow us to specify where the XSD is located itself, I think we should keep it

File: podcast.xsd [hunk 2]
```diff
@@ -18,6 +16,83 @@
     </xsd:documentation>
 </xsd:annotation>
 
+<!-- Custom type definition -->
+<xsd:simpleType name="RestrictYesNo">
+	<xsd:restriction base="xsd:string">
```
Comment (PofMagicfingers) [inline podcast.xsd:21]: Nice!

File: podcast.xsd [hunk 3]
```diff
@@ -18,6 +16,83 @@
...
+		<xsd:enumeration value="yes" />
+		<xsd:enumeration value="no" />
+	</xsd:restriction>
+</xsd:simpleType>
+
+<xsd:simpleType name="Platform">
+	<xsd:restriction base="xsd:string">
```
Comment (PofMagicfingers) [inline podcast.xsd:28]: Is a Platform type really needed if it's the same restrictions as a Name ? After all, it just a platform name ^^

File: podcast.xsd [hunk 4]
```diff
@@ -18,6 +16,83 @@
...
+<xsd:simpleType name="LinkHref">
+	<xsd:restriction base="xsd:anyURI">
+		<xsd:minLength value="12"/>
+		<xsd:maxLength value="2048"/>
+	</xsd:restriction>
+</xsd:simpleType>
+
+<xsd:simpleType name="ISBN">
+    <xsd:restriction base="xsd:string">
+		<xsd:pattern value="\d{1}-\d{5}-\d{3}-\d{1}|\d{1}-\d{3}-\d{5}-\d{1}|\d{1}-\d{2}-\d{6}-\d{1}"/>
```
Comment (PofMagicfingers) [inline podcast.xsd:50]: :+1:

File: podcast.xsd [hunk 5]
```diff
@@ -18,6 +16,83 @@
...
+	<xsd:annotation>
+		<xsd:documentation>
+			This tag give some information about person in a podcast
+		</xsd:documentation>
+	</xsd:annotation>
+	<xsd:choice minOccurs="0" maxOccurs="unbounded">
+			<xsd:element name="link">
+				<xsd:annotation>
+					<xsd:documentation>
+						Used to add any link you want related to this people podcast.
```
Comment (PofMagicfingers) [inline podcast.xsd:64]: "to this people podcast." => not sure what that means

File: podcast.xsd [hunk 6]
```diff
@@ -18,6 +16,83 @@
...
+				<xsd:complexType>
+					<xsd:simpleContent>
+						<xsd:extension base="LinkHref">
+							<xsd:attribute name="href" type="LinkHref" use="required"/>	
+						</xsd:extension>
+					</xsd:simpleContent>
+				</xsd:complexType>
+			</xsd:element>
+
+			<xsd:element name="social">
```
Comment (PofMagicfingers) [inline podcast.xsd:76]: `podcast:social` tags will also be available at channel and item level. I think this should be defined outside of `Person` and referenced here and there.

File: podcast.xsd [hunk 7]
```diff
@@ -136,15 +190,16 @@
...
-    <xsd:simpleType>
-        <xsd:restriction base="xsd:string">
-            <xsd:enumeration value="yes" />
-            <xsd:enumeration value="no" />
-        </xsd:restriction>
-    </xsd:simpleType>
+	<xsd:complexType>
+		<xsd:simpleContent>
+			<xsd:extension base="RestrictYesNo">
+				<xsd:attribute name="platforme" type="Platform" use="required"/>
```
Comment (PofMagicfingers) [inline podcast.xsd]:
```suggestion
				<xsd:attribute name="platform" type="Platform" use="required"/>
```

File: podcast.xsd [hunk 8]
```diff
@@ -136,15 +190,16 @@
...
+			</xsd:extension>
+		</xsd:simpleContent>
+	</xsd:complexType>	
 </xsd:element>
 
-<xsd:element name="newFeedUrl">
+<xsd:element name="newFeedUrl" type="LinkHref">
```
Comment (PofMagicfingers) [inline podcast.xsd:202]: not really sure we should include that kind of tags. I guess I forgot to remove it when I used google play xsd as work base

File: podcast.xsd [hunk 9]
```diff
@@ -176,4 +225,111 @@
     </xsd:complexType>
 </xsd:element>
 
+<xsd:element name="social">
```
Comment (PofMagicfingers) [inline podcast.xsd:228]: Yep, that one social. We need to reference it on Person type. I can't find the doc where they explain how to reference type inside another...

File: podcast.xsd [hunk 10]
```diff
@@ -176,4 +225,111 @@
...
+	<xsd:complexType>
+		<xsd:simpleContent>
+			<xsd:extension base="LinkHref">
+				<xsd:attribute name="href" type="LinkHref" use="required"/>	
+			</xsd:extension>
+		</xsd:simpleContent>
+	</xsd:complexType>
+</xsd:element>
+
+<xsd:element name="person" type="Person" />
```
Comment (PofMagicfingers) [inline podcast.xsd:282]: Okay that was that. I think we have to create a `PlatformLink` type with :
```
	<xsd:complexType>
		<xsd:attribute name="platform" type="Platform" use="required" />
		<xsd:attribute name="handle" type="Name" use="optional" />
		<xsd:attribute name="href" type="LinkHref" use="required" />
	</xsd:complexType>
```

and use : `<xsd:element name="social" type="Social" />` in Person and here.

File: podcast.xsd [hunk 11]
```diff
@@ -176,4 +225,111 @@
...
+	</xsd:annotation>
+
+	<xsd:complexType>
+		<xsd:choice minOccurs="1" maxOccurs="unbounded">
+			<xsd:element name="person" type="Person" />
+		</xsd:choice>
+	</xsd:complexType>
+</xsd:element>
+
+<xsd:element name="guest">
```
Comment (PofMagicfingers) [inline podcast.xsd]:
```suggestion
<xsd:element name="guests">
```

**Review by PofMagicfingers — CHANGES_REQUESTED** [review]
A few things to change, but it's a big first step! Thanks a lot!

File: podcast.xsd (hunk 3 above)
Comment (Bigaston) [reply podcast.xsd:28]: Yes, that's true!

File: podcast.xsd (hunk 5 above)
Comment (Bigaston) [reply podcast.xsd:64]: Yes my english is not perfect xD

File: podcast.xsd (hunk 8 above)
Comment (Bigaston) [reply podcast.xsd:202]: Okay!

File: podcast.xsd (hunk 7 above)
Comment (Bigaston) [reply podcast.xsd]: Woups! Sorry ^^'

File: podcast.xsd (hunk 9 above)
Comment (Bigaston) [reply podcast.xsd:228]: I have not find it to... When I need to specify that team need to include only person I've find how but for that... nothing

File: podcast.xsd (hunk 10 above)
Comment (Bigaston) [reply podcast.xsd:282]: Yes!

File: podcast.xsd (hunk 5 above)
Comment (PofMagicfingers) [reply podcast.xsd:64]: What did you try to say ? :upside_down_face:

**Pull request merged**